# Annotating genomes the Ensembl way: Hands on - From RNAseq reads to gene models

This session is part of [**Biodiversity Genomics Academy 2023**](https://BGA23.org)

## Session Leader(s)

Jose Maria Gonzalez Perez-Silva
Bioinformatician
EMBL-EBI

## Description

By the end of this session you will be able to:
1. Perform quality control and pre-processing of RNA-Seq reads: This includes trimming of adapter sequences and quality control to ensure that the reads are of sufficient quality to be used for analysis.
2. Align RNA-Seq reads to the genome: The reads are aligned to the genome assembly, allowing for the identification of expressed genes and transcripts.
3. Assemble transcripts: The aligned reads are used to reconstruct the transcripts, or messenger RNAs (mRNAs), that are expressed in the sample.
4. Annotate genes: The identified genes can be annotated using a combination of homology-based and ab initio gene prediction methods. The homology-based methods use existing gene information from closely related species, while the ab initio methods use the transcript assembly information to predict new genes.

## Prerequisites

1. Understand the terms genome assembly, reads, contigs
2. Understand what a gene model is, what introns and exons are
3. Understand what transcriptomic and protein data are, including understanding the different types of RNA
4. Understand the concept of sequence alignment

!!! warning "Please make sure you MEET THE PREREQUISITES and READ THE DESCRIPTION above"

    You will get the most out of this session if you meet the prerequisites above.

    Please also read the description carefully to see if this session is relevant to you.
    
    If you don't meet the prerequisites or change your mind based on the description or are no longer available at the session time, please email tol-training at sanger.ac.uk to cancel your slot so that someone else on the waitlist might attend.
